### ISSUE TITLE
Cannot modify ContentDialog parameters in winui 2.6 prelease

### ISSUE THREAD

**Issue by lhak** [microsoft/microsoft-ui-xaml#4247]
**Describe the bug**

When setting the Version parameter of the XamlControlsResources object to "latest", it is no longer possible to modify the ContentDialog style

**Steps to reproduce the bug**

1. Create a new winui 2.6 app and change the version to "latest"
2. Add a ContentDialog to the project
3. Try to modify its padding by changing it in the resources section:
```xaml
<ContentDialog.Resources>
   <Thickness x:Key="ContentDialogPadding">0</Thickness>
</ContentDialog.Resources>
```
4. Observe that the padding does not change

**Expected behavior**

Padding should change

**Screenshots**

**Version Info**

NuGet package version: 

[Microsoft.UI.Xaml.2.6.0-prerelease.210217002]

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| Yes |  |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| October 2020 Update (19042)        | Yes  |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

**Comment by StephenLPeters** [comment]
@lhak Is this the first content dialog your app has shown? I suspect that what is happening is this value is now being referenced by a StaticResource markup extension instead of a themeResource markup extension. StaticResource extensions are only resolved once. I'm guessing you have another ContentDialog that shows before this one that locks in the default value and then your override is not getting applied.

**Comment by StephenLPeters** [comment]
The fix here is to change this https://github.com/microsoft/microsoft-ui-xaml/blob/92dcdc0c6bd97dca35d248f13a930aec02e0d105/dev/ContentDialog/ContentDialog_themeresources.xaml#L283 and https://github.com/microsoft/microsoft-ui-xaml/blob/92dcdc0c6bd97dca35d248f13a930aec02e0d105/dev/ContentDialog/ContentDialog_themeresources.xaml#L235 to use the theme resource extension. We should also move the resource itself into the theme dictionaries.

**Comment by lhak** [comment]
The Expander control has the same issue

**Comment by tipa** [comment]
I have the same problem.
Any update? Why is this issue stale for so many months already?
Sounds like this could be easily fixed.

**Comment by michael-hawker** [comment]
@StephenLPeters why is making a value a `ThemeResource` the solution here? Shouldn't all padding/margin fixed values just be static resources? They should still be overridable by the initial setting of a Style (or is this a workaround to some other issue?). ThemeResources and the Theme dictionary should just be for brushes which need to be updated when the app theme changes from light/dark/high contract, no?

**Comment by chingucoding** [comment]
@michael-hawker That's because StaticResources get evaluated once only, however that one time is too early for the lightweight styling resources to take effect. That's why using a ThemeResource binding which handles this a bit differently will result in the resource pickup working again.

**Comment by michael-hawker** [comment]
@chingucoding thanks for the explanation but that seems like another bug then? As long as I've defined a resource override up the visual tree, it should be available to the control when it applies the template?

We don't want to define every value as a `ThemeResource` as otherwise what's the point of a `StaticResource`?

**Comment by chingucoding** [comment]
This is sort of a bug agreed @michael-hawker. The thing is that if you don't import external resources and styles, using StatisResource should work fine. I think the issue with WinUI is just how it loads resources and the fact that it's loading external styles.

**Comment by michael-hawker** [comment]
Fixed by #5834 for #5688.
